src/services/db/RepoService.ts

```diff
@@ -144,6 +151,51 @@ export default class RepoService extends GitHubService {
     })
   }
 
+  async readMediaFile(
+    sessionData: UserWithSiteSessionData,
+    { fileName, directoryName }: { fileName: string; directoryName: string }
+  ): Promise<any> {
+    logger.debug(`Reading media file: ${fileName}`)
+    logger.debug(`Reading directoryName: ${directoryName}`)
+    const { siteName } = sessionData
+
+    // fetch from local disk
+    if (this.isRepoWhitelisted(siteName)) {
+      logger.info(
+        `Reading media file from disk. Sitname: ${siteName}, directory name: ${directoryName}, fileName: ${fileName},`
+      )
+      const result = await this.gitFileSystemService.readMediaFile(
+        siteName,
+        directoryName,
+        fileName
+      )
+      if (result.isErr()) {
+        throw result.error
+      }
+
+      return result.value
+    }
+
+    // fetch from Github
+    const directoryData = await super.readDirectory(sessionData, {
+      directoryName,
+    })
+
+    const mediaType = directoryName.split("/")[0] as MediaType
+    const targetFile = directoryData.find(
+      (fileOrDir: { name: string }) => fileOrDir.name === fileName
+    )
+    const { private: isPrivate } = await super.getRepoInfo(sessionData)
```

Comment:
Similarly, should we do a `try catch` to handle potential error from calling `super.getRepoInfo` here?